sdk/core/Azure.Core/src/Serialization/IJsonModelSerializable.cs

```diff
@@ -0,0 +1,32 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Text.Json;
+
+namespace Azure.Core.Serialization
+{
+    /// <summary>
+    /// .
+    /// </summary>
+    public interface IJsonModelSerializable : IModelSerializable
+    {
+        /// <summary>
+        /// .
+        /// </summary>
+        /// <param name="writer"></param>
+        /// <param name="options"></param>
+#pragma warning disable AZC0014 // Avoid using banned types in public API
+        void Serialize(Utf8JsonWriter writer, ModelSerializerOptions options);
```

Comment:
Curious: did we consider async overloads?  If so, what made us decide against them?

Reply:
no we had no considered async overloads but I think we probably should